File: backend/src/main/java/com/octo/docstack/service/ProfileService.java
package com.octo.docstack.service;

import com.octo.docstack.dto.profile.ProfileCreateRequest;
import com.octo.docstack.dto.profile.ProfileResponse;
import com.octo.docstack.dto.profile.ProfileUpdateRequest;

public interface ProfileService {

    ProfileResponse getMyProfile(String clerkUserId);

    ProfileResponse syncMyProfileFromClerk(String clerkUserId);

    ProfileResponse createProfile(ProfileCreateRequest request);

    ProfileResponse getProfileByClerkUserId(String clerkUserId);

    ProfileResponse updateProfile(String clerkUserId, ProfileUpdateRequest request);

    void deleteProfile(String clerkUserId);
}

File: frontend/services/profile/profileService.ts
import { apiFetch } from "@/lib/api";

export interface Profile {
  clerkUserId: string;
  firstName: string;
  lastName: string;
  email: string;
}

export const profileService = {
  createProfile(token: string) {
    return apiFetch("/api/profile/me/sync", { method: "POST" }, token);
  },

  getMyProfile(token: string) {
    return apiFetch<Profile>("/api/profile/me", { method: "GET" }, token);
  },

  updateProfile(token: string, data: { firstName: string; lastName: string }) {
    return apiFetch<Profile>(
      "/api/profile",
      {
        method: "PATCH",
        body: JSON.stringify(data),
      },
      token
    );
  },

  deleteProfile(token: string) {
    return apiFetch<void>("/api/profile", { method: "DELETE" }, token);
  },
};
